### PR TITLE
Adjust HOC to support Next.js v6

### PR DESCRIPTION
**Pull request by RobbinHabermehl**
While there's some work left to do, this PR fixes the incompatibility with Next.js v6.

To-do's:
- [x] Fix unit tests
- [x] Update `README`

I'm using the following code to utilize `next-redux-saga` in my app.

`pages/_app.js`:
```js
import React from 'react';
import withRedux from 'next-redux-wrapper';

import nextReduxSaga from '../next-redux-saga';
import initializeStore from '../store';

class MyApp extends React.Component {
  static async getInitialProps({ Component, ctx }) {
    ctx.store.dispatch({ type: 'FOO', payload: 'foo' });

    const pageProps = Component.getInitialProps ? await Component.getInitialProps(ctx) : {};

    return { pageProps };
  }

  render() {
    const { Component, pageProps } = this.props;
    return (
      <Component {...pageProps} />
    );
  }
}

export default withRedux(initializeStore)(nextReduxSaga(MyApp));
```

Please note that I've got a copy of the source locally as obviously it hasn't been released yet:

`next-redux-saga.js`:
```js
import React, {Component} from 'react'
import {END} from 'redux-saga'

function hoc(config) {
  return BaseComponent => {
    class WrappedComponent extends Component {
      static async getInitialProps(props) {
        const {isServer, store} = props.ctx

        let pageProps = {}
        if (BaseComponent.getInitialProps) {
          pageProps = await BaseComponent.getInitialProps(props)
        }

        // Keep saga running on the client (async mode)
        if (config.async && !isServer) {
          return pageProps
        }

        // Force saga to end in all other cases
        store.dispatch(END)
        await store.sagaTask.done

        // Restart saga on the client (sync mode)
        if (!isServer) {
          store.runSagaTask()
        }

        return pageProps
      }

      render() {
        return <BaseComponent {...this.props} />
      }
    }

    return WrappedComponent
  }
}

function withReduxSaga(arg) {
  const defaultConfig = {async: false}

  if (typeof arg === 'function') {
    return hoc(defaultConfig)(arg)
  }

  return hoc({...defaultConfig, ...arg})
}

export default withReduxSaga
```

### CODE REVIEW

**Comment by bmealhouse** [comment]
@RobbinHabermehl - I started looking at this PR today. Your changes look good, but the tests and example app will need a lot of changes to work properly with Next.js v6 and `next-redux-wrapper`. I should be able to help out with the tests and example app tomorrow and get this ready for release.

**Comment by RobbinHabermehl** [comment]
Yeah, I noticed @bmealhouse. I put in all the time I had on Friday and focused on getting it to work. I see you've made good progress though! Would you like me to assist on anything?

**Comment by lukebarton** [comment]
Hey guys! Thanks for sorting this out. Will it get merged any time soon? Currently blocked - need to decide what to do.

**Comment by bmealhouse** [comment]
Yes, we just need to get the docs updated and then I can get it released. I think it’s realistic to have this released tonight.

File: README.md [hunk 1]
```diff
@@ -64,14 +66,47 @@ function configureStore(initialState) {
...
+      <Container>
+        <Provider store={store}>
+          <Component {...pageProps} />
+        </Provider>
+      </Container>
+    )
+  }
+}
+
+export default withRedux(configureStore)(withReduxSaga(ExampleApp))
```
Comment (prtkkmrsngh) [inline README.md]: `export default withRedux(configureStore)(nextReduxSaga(ExampleApp))`

isn't it?

**Comment by RobbinHabermehl** [comment]
Great job @bmealhouse, thanks for taking care of the remaining work!

**Comment by bmealhouse** [comment]
@prtkkmrsngh Yes, you are correct. Thanks for pointing that out. Naming it `nextReduxSaga` will still work as long as it's imported with the same name. In any case, I have some more updates to make to the repository, so I will update this documentation as well.

**Comment by bmealhouse** [comment]
@RobbinHabermehl No problem! Thanks again for getting the PR started.

**Pull request merged**